File: app/src/main/java/ch/unstable/ost/lists/query/QueryBinder.java
package ch.unstable.ost.lists.query;


import android.content.Context;
import android.support.annotation.MainThread;
import android.widget.TextView;

import com.google.common.base.Preconditions;

import java.util.Date;

import ch.unstable.ost.R;
import ch.unstable.ost.api.model.Connection;
import ch.unstable.ost.api.model.ConnectionQuery;
import ch.unstable.ost.database.model.QueryHistory;
import ch.unstable.ost.utils.LocalizationUtils;

public class QueryBinder {

    @MainThread
    public static void bindQuery(QueryHistory queryEntry, TextView dateView, TextView fromAndTo) {
        final Context context = dateView.getContext();
        final ConnectionQuery query = queryEntry.getQuery();
        Date arrival = query.getArrivalTime();
        Date departure = query.getDepartureTime();
        if (query.isNow()) {
            // The search was for "now" so we show the time the query was submitted
            departure = queryEntry.getCreationDate();
        }
        dateView.setText(LocalizationUtils.getArrivalOrDepartureText(context, arrival, departure));
        bindFromToText(fromAndTo, query);
    }

    public static void bindFromToText(TextView textView, ConnectionQuery query) {
        //noinspection ResultOfMethodCallIgnored
        Preconditions.checkNotNull(query, "query is null");
        //noinspection ResultOfMethodCallIgnored
        Preconditions.checkNotNull(textView, "textView is null");
        textView.setText(textView.getContext().getString(R.string.fromAndTo, query.getFrom(), query.getTo()));
    }

    public static void bindFromToText(TextView textView, Connection connection) {
        //noinspection ResultOfMethodCallIgnored
        Preconditions.checkNotNull(connection, "connection is null");
        //noinspection ResultOfMethodCallIgnored
        Preconditions.checkNotNull(textView, "textView is null");
        textView.setText(textView.getContext().getString(R.string.fromAndTo, connection.getDepartureName(), connection.getArrivalName()));
    }
}

File: app/src/main/java/ch/unstable/ost/views/lists/query/QueryBinder.kt
package ch.unstable.ost.views.lists.query


import android.support.annotation.MainThread
import android.widget.TextView
import ch.unstable.ost.R
import ch.unstable.ost.api.model.Connection
import ch.unstable.ost.api.model.ConnectionQuery
import ch.unstable.ost.database.model.QueryHistory
import ch.unstable.ost.utils.LocalizationUtils

object QueryBinder {

    @MainThread
    fun bindQuery(queryEntry: QueryHistory, dateView: TextView, fromAndTo: TextView) {
        val query = queryEntry.query
        // The search was for "now" so we show the time the query was submitted
        val departure = if (query.isNow) query.departureTime else queryEntry.creationDate
        dateView.text = LocalizationUtils.getArrivalOrDepartureText(dateView.context, query.arrivalTime, departure)
        bindFromToText(fromAndTo, query)
    }

    fun bindFromToText(textView: TextView, query: ConnectionQuery) {
        textView.text = textView.context.getString(R.string.fromAndTo, query.from, query.to)
    }

    fun bindFromToText(textView: TextView, connection: Connection) {
        textView.text = textView.context.getString(R.string.fromAndTo, connection.departureName, connection.arrivalName)
    }
}
